pkg/admissioncontroller/webhooks/auth/seed/handler.go

```diff
@@ -0,0 +1,232 @@
+	defer cancel()
+
+	decision, reason, err := h.authorizer.Authorize(ctx, AuthorizationAttributesFrom(*sarSpec))
+	if err != nil {
+		h.logger.Error(err, "error when consulting authorizer for opinion")
+		h.writeResponse(w, gvk, Errored(http.StatusInternalServerError, err))
+		return
+	}
+
+	var status authorizationv1beta1.SubjectAccessReviewStatus
+	switch decision {
+	case authorizer.DecisionAllow:
+		status = Allowed()
+	case authorizer.DecisionDeny:
+		status = Denied(reason)
+	case authorizer.DecisionNoOpinion:
+		status = NoOpinion(reason)
+	default:
+		status = Errored(http.StatusInternalServerError, fmt.Errorf("unexpected decision: %d", decision))
+	}
+
+	h.logger.V(1).Info("responding to request", append([]interface{}{"decision", decision, "reason", reason}, keysAndValues...))
+	h.writeResponse(w, gvk, status)
+}
+
+func (h *handler) writeResponse(w io.Writer, gvk *schema.GroupVersionKind, status authorizationv1beta1.SubjectAccessReviewStatus) {
+	// The SubjectAccessReviewStatus type is exactly the same for both authorization.k8s.io/v1beta1 and
+	// authorization.k8s.io/v1, so we can just overwrite the apiVersion/kind with the same version like the object in
+	// the request.
+	response := authorizationv1beta1.SubjectAccessReview{Status: status}
+	if gvk == nil || *gvk == (schema.GroupVersionKind{}) {
+		response.SetGroupVersionKind(authorizationv1beta1.SchemeGroupVersion.WithKind("SubjectAccessReview"))
+	} else {
+		response.SetGroupVersionKind(*gvk)
+	}
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		h.logger.Error(err, "unable to encode the response")
+		h.writeResponse(w, gvk, Errored(http.StatusInternalServerError, err))
+		return
+	}
+
+	if log := h.logger; log.V(1).Enabled() {
+		log = log.WithValues(
+			"allowed", status.Allowed,
+			"denied", status.Denied,
+			"reason", status.Reason,
+			"error", status.EvaluationError,
+		)
+		log.V(1).Info("wrote response")
+	}
+}
+
+func (h *handler) decodeRequestBody(body []byte) (*authorizationv1beta1.SubjectAccessReviewSpec, *schema.GroupVersionKind, error) {
+	var (
+		obj     unstructured.Unstructured
+		sarSpec authorizationv1beta1.SubjectAccessReviewSpec
+	)
+
+	_, gvk, err := codecs.UniversalDeserializer().Decode(body, nil, &obj)
+	if err != nil {
+		return nil, nil, err
+	}
+	if gvk == nil {
+		return nil, nil, fmt.Errorf("could not determine GVK for object in the request body")
+	}
+
+	switch *gvk {
+	case authorizationv1beta1.SchemeGroupVersion.WithKind("SubjectAccessReview"):
+		var tmp authorizationv1beta1.SubjectAccessReview
+		if err := scheme.Convert(&obj, &tmp, nil); err != nil {
+			return nil, nil, err
+		}
+		return &tmp.Spec, gvk, nil
+
+	case authorizationv1.SchemeGroupVersion.WithKind("SubjectAccessReview"):
+		var tmp authorizationv1.SubjectAccessReview
+		if err := scheme.Convert(&obj, &tmp, nil); err != nil {
+			return nil, nil, err
+		}
+
+		// "convert" authorizationv1.SubjectAccessReviewSpec to authorizationv1beta1.SubjectAccessReviewSpec
```

Comment:
How about if we leave it like before i.e., always decode to `authorizationv1` and if the API server sent `authorizationv1beta1`, decode it to `authorizationv1beta1` + only feed the `sarSpec.Groups` portion (because that's the difference between the two API). Maybe I'm on the wrong track but I think this would spare us some boilerplate?! Anyways, please consider it as a nit. I'm also fine with the current handling.